.github/workflows/main.yml

```diff
@@ -207,3 +207,49 @@ jobs:
         with:
           command: clippy
           args: -- -D warnings
+
+  test_coverage:
+    name: Test Coverage
+    needs: update_deps
+    runs-on: ubuntu-latest
+    env:
+      SHA_SHORT: ${GITHUB_SHA:0:8}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Download Cargo.lock
+        uses: actions/download-artifact@v2
+        with:
+          name: lockfile
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Restore cargo registry index
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/registry/index
+          key: cargo-index-v2-${{ needs.update_deps.outputs.crates-io-index-head }}
+
+      - name: Restore dependency crates
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/registry/cache
+          key: cargo-deps-v2-${{ hashFiles('Cargo.lock') }}
+
+      - name: Run tarpaulin
+        uses: actions-rs/tarpaulin@v0.1
+        with:
+          out-type: Html
+
+      - run: mv tarpaulin-report.html test_coverage_$SHA_SHORT.html
```

Comment:
I would reflect a branch name & commit time as well. It would be easier to navigate through artifacts that way.

Reply:
You can actually drop this extra step by doing

```yaml
- uses: actions/upload-artifact@v2
  with:
    name: test_coverage_${{ env.SHA_SHORT }}
    path: tarpaulin-report.html
```

Reply:
The artefact is attached to the workflow run, which has the links to GitHub.

Reply:
@eugene-babichenko I believe that skipping the `mv` would leave the actual file name as is, meaning all the reports would be `tarpaulin-report.html` after unzipping. That defeats the purpose of including the SHA and branch name in the artifact, don't you think?

@mzabaluev I'm not sure what you mean. The idea with the naming is to be able to distinguish them on our local filesystems or in case they're shuffled around over Slack, email, and whatnot. 

Branch name would be nice to have 👌 I'd replace any character that is not alphanumeric, `_`, or `-` with `-` for a cleaner name. I'm not so keen on the timestamp since it makes the file name quite verbose – unless it's something compact. For example:
```
test_coverage_feature-branch-name_20210713T112706Z_3df40ba2.html
```

What do you think?